lib/scala/interpreter-dsl/src/main/java/org/enso/interpreter/dsl/BuiltinsMetadataProcessor.java

```diff
@@ -0,0 +1,72 @@
+package org.enso.interpreter.dsl;
+
+import org.enso.interpreter.dsl.model.MethodDefinition;
+
+import javax.annotation.processing.AbstractProcessor;
+import javax.annotation.processing.Filer;
+import javax.annotation.processing.RoundEnvironment;
+import javax.lang.model.element.TypeElement;
+import javax.tools.FileObject;
+import javax.tools.StandardLocation;
+import java.io.IOException;
+import java.io.Writer;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+public abstract class BuiltinsMetadataProcessor extends AbstractProcessor {
```

Comment:
Please make sure this plays nicely with the incremental compiler (i.e. the metadata does not get trashed when you add a new builtin)

Reply:
(I'm not saying it doesn't, all I'm saying is "I've been hurt before, please make sure I'm not hurt this time")

Reply:
Yeah, it is a problem atm. I might have to revert this writing to a single metada file because of it. 
I will see if there is a way around it /cc @JaroslavTulach